### PR TITLE
Features/stepper

### PR DESCRIPTION
**Pull request by GoodVincentTu**

### CODE REVIEW

File: src/components/AuctionStepper/AuctionStepper.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+//           </Step>
+//           <Step>
+//             <StepLabel>Finalize</StepLabel>
+//           </Step>
+//         </Stepper>
+//       </div>
+//     );
+//   };
+// };
+
```
Comment (chochinlu) [inline src/components/AuctionStepper/AuctionStepper.js]: (Suggestion)  If we deleted a block of codes, we  can commit it.  That can make the all code look clean and we can rollback our code by git.

File: src/components/AuctionStepper/AuctionStepper.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+const Step = (props) => (
+  // TODO
+  // this is for temporary testing step change
+  <div onClick={()=>{props.setStep(props.step)}}>
```
Comment (chochinlu) [inline src/components/AuctionStepper/AuctionStepper.js]:
``` jsx
<div onClick={()=> props.setStep(props.step) }>
```

File: src/components/AuctionStepper/AuctionStepper.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+  return (
+    <div>
+      {steps.map((step, index) => 
+        <Step
+          key={`step-${index}`}
+          index={index}
+          title={step.title}
+          step={step.name}
+          setStep={props.setStep}
+        />
```
Comment (chochinlu) [inline src/components/AuctionStepper/AuctionStepper.js]:
``` jsx
<Step
 +          key={`step-${index}`}
 +          index={index}
 +          {...step}
 +          setStep={props.setStep}
 +        />
```

File: src/components/MainWrapper.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import {FAQ} from './FAQ';
+import {About} from './About';
+import {AuctionWrapper} from './AuctionWrapper/AuctionWrapper';
+import './MainWrapper.css';
+
+const Main = (props) => (
+  <div>
+    <SearchEns
+      switchPage={props.switchPage} />
+    <hr/>
```
Comment (chochinlu) [inline src/components/MainWrapper.js]: `<hr />`  can be replaced by CSS.

For example, in `<SearchEns>`,  we can set CSS style:  `border-bottom: 1px solid black`

File: src/lib/util.js [hunk 5]
```diff
@@ -11,4 +11,18 @@ export const Box = x => ({
...
+  if (urlSearchString.indexOf('?') == -1) return false;
+  urlSearchString = urlSearchString.split('?')[1];
+  const query = urlSearchString
+    .split('&')
+    .map(q => q.split('='))
+    .filter(([name, value]) => name && value)
+    .map(([name, value]) => {name: decodeURI(value)})
+    .reduce((a, b) => Object.assign(a,b), {});
+  return query;
+}
```
Comment (chochinlu) [inline src/lib/util.js]: COOL!   We can also write a test for this function to ensure it work.

Comment (GoodVincentTu) [reply src/lib/util.js]: yes, i shell put the simple test later.

**Pull request merged**